### PR TITLE
Bug 2070744: Fix infinite loop error

### PR DESCRIPTION
**Pull request by rna-afk**
There is an infinite loop if the resource is not untagged during AWS destroy. This needs to be checked for an InvalidParameterException and if it is, we should not retry the same client again.

### CODE REVIEW

File: pkg/destroy/aws/shared.go [hunk 1]
```diff
@@ -117,6 +117,12 @@ func (o *ClusterUninstaller) removeSharedTag(ctx context.Context, session *sessi
 				}
 				result, err := tagClient.UntagResourcesWithContext(ctx, request)
 				if err != nil {
+					if aerr, ok := err.(awserr.Error); ok {
```
Comment (r4f4) [inline pkg/destroy/aws/shared.go:120]: We should probably be using `errors.As` so typecasting can work even with wrapped errors.

File: pkg/destroy/aws/shared.go [hunk 2]
```diff
@@ -117,6 +117,12 @@ func (o *ClusterUninstaller) removeSharedTag(ctx context.Context, session *sessi
...
+						switch aerr.Code() {
+						case resourcegroupstaggingapi.ErrorCodeInvalidParameterException:
+							nextTagClients = nextTagClients[:len(nextTagClients)-1]
```
Comment (patrickdillon) [inline pkg/destroy/aws/shared.go:123]: I'm not following this line. Why do this? Can you add a comment in the code?

Comment (rna-afk) [reply pkg/destroy/aws/shared.go:128]: turns out I was right the first time. tagClient is added to the last of the list and has to be removed if IPE is returned.

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Pull request merged**